queue_job/jobrunner/runner.py

```diff
@@ -311,6 +350,11 @@ def _has_queue_job(self):
 
     def _initialize(self):
         with closing(self.conn.cursor()) as cr:
+            if self.jobrunner_ha_uuid:
+                cr.execute(
+                    "SET idle_in_transaction_session_timeout TO %s",
+                    ((IDLE_TRANSACTION_TIMEOUT * 1000,)),
+                )
```

Comment:
Why is this necessary? Since the connection is in autocommit mode, there should be no transaction?

Reply:
This was added at the suggestion of Guewen. My understanding was that it is intended as a safeguard for edge cases or unanticipated states that could leave a transaction idle thus blocking the queue job processing.

Reply:
Retrospectively, I don't know why I proposed this 😓 it has no effect outside of a transaction